forge/src/multi_runner.rs

```diff
@@ -267,88 +267,283 @@ mod tests {
+                    vec![
+                        (
+                            "testInts()",
+                            true,
+                            None,
+                            Some(vec![
+                                "constructor".into(),
+                                "0".into(),
+                                "1".into(),
+                                "2".into(),
+                                "3".into(),
+                            ]),
+                        ),
+                        (
+                            "testMisc()",
+                            true,
+                            None,
+                            Some(vec![
+                                "constructor".into(),
+                                "testMisc, 0x0000000000000000000000000000000000000001".into(),
+                                "testMisc, 42".into(),
+                            ]),
+                        ),
+                        (
+                            "testStrings()",
+                            true,
+                            None,
+                            Some(vec!["constructor".into(), "testStrings".into()]),
+                        ),
+                    ],
+                ),
+            ]),
+        );
     }
 
     #[test]
-    fn test_abstract_contract() {
+    fn test_cheats() {
         let mut runner = runner();
-        let results = runner.test(&Filter::new(".*", ".*", ".*"), None).unwrap();
-        assert!(results.get("Tests.json:Tests").is_none());
-        assert!(results.get("ATests.json:ATests").is_some());
-        assert!(results.get("BTests.json:BTests").is_some());
+        let results = runner.test(&Filter::new(".*", ".*", ".*cheats"), None).unwrap();
+
+        for (_, tests) in results {
+            for (test_name, result) in tests {
+                assert!(
+                    result.success,
+                    "Test {} did not pass as expected.\nReason: {:?}",
+                    test_name, result.reason
+                );
+            }
+        }
     }
 
     #[test]
-    fn test_debug_logs() {
+    fn test_fuzz() {
         let mut runner = runner();
-        let results = runner.test(&Filter::new(".*", ".*", ".*"), None).unwrap();
+        let results = runner.test(&Filter::new(".*", ".*", ".*fuzz"), None).unwrap();
+
+        for (_, tests) in results {
+            for (test_name, result) in tests {
+                match test_name.as_ref() {
+                    "testPositive(uint256)" | "testSuccessfulFuzz(uint128,uint128)" => assert!(
+                        result.success,
+                        "Test {} did not pass as expected.\nReason: {:?}",
+                        test_name, result.reason
+                    ),
+                    _ => assert!(
+                        !result.success,
+                        "Test {} did not fail as expected.\nReason: {:?}",
+                        test_name, result.reason
+                    ),
```

Comment:
Wonder if we should add some more fuzzing unit tests that show the success case, given here I think we have a lot of "failign" ones which we use to gain confidence